plugins/org.jboss.tools.openshift.core/src/org/jboss/tools/openshift/core/connection/Connection.java

```diff
@@ -314,13 +314,32 @@ private IAuthorizationStrategy getAuthorizationStrategy() {
 	 * @return
 	 */
 	public boolean isConnected(IProgressMonitor monitor) {
-		client.setAuthorizationStrategy(getAuthorizationStrategy());
+		boolean needStrategy = initAuthorizationStrategy();
 		try {
 			IAuthorizationContext context = client.getContext(client.getBaseURL().toString());
 			return context.isAuthorized();
 		} catch (UnauthorizedException e) {
+			needStrategy = false;
 			return false;
+		} finally {
+			if(needStrategy) {
+				connect();
```

Comment:
this causes ConnectionTest#getResourceKindShouldCallClient() to fail with an NPE:
```
java.lang.NullPointerException
	at org.jboss.tools.openshift.core.connection.Connection.authorize(Connection.java:270)
	at org.jboss.tools.openshift.core.connection.Connection.connect(Connection.java:259)
	at org.jboss.tools.openshift.core.connection.Connection.getResources(Connection.java:482)
	at org.jboss.tools.openshift.core.connection.Connection.getResources(Connection.java:468)
	at org.jboss.tools.openshift.test.core.connection.ConnectionTest.getResourceKindShouldCallClient(ConnectionTest.java:97)
```